include/net/ethernet.h

```diff
@@ -43,15 +47,54 @@ extern "C" {
 
 #define NET_ETH_MINIMAL_FRAME_SIZE	60
 
-enum eth_hw_caps {
+enum ethernet_hw_caps {
 	/** TX Checksum offloading supported */
-	ETH_HW_TX_CHKSUM_OFFLOAD  = BIT(0),
+	ETHERNET_HW_TX_CHKSUM_OFFLOAD	= BIT(0),
 
 	/** RX Checksum offloading supported */
-	ETH_HW_RX_CHKSUM_OFFLOAD  = BIT(1),
+	ETHERNET_HW_RX_CHKSUM_OFFLOAD	= BIT(1),
 
 	/** VLAN supported */
-	ETH_HW_VLAN  = BIT(2),
+	ETHERNET_HW_VLAN		= BIT(2),
+
+	/** Enabling/disabling auto negotiation supported */
+	ETHERNET_AUTO_NEGOTIATION_SET	= BIT(3),
+
+	/** 10 Mbits link supported */
+	ETHERNET_LINK_10BASE_T		= BIT(4),
+
+	/** 100 Mbits link supported */
+	ETHERNET_LINK_100BASE_T		= BIT(5),
+
+	/** 1 Gbits link supported */
+	ETHERNET_LINK_1000BASE_T	= BIT(6),
+
+	/** Changing duplex (half/full) supported */
+	ETHERNET_DUPLEX_SET		= BIT(7),
+};
+
+enum ethernet_config_type {
+	ETHERNET_CONFIG_TYPE_AUTO_NEG,
+	ETHERNET_CONFIG_TYPE_LINK,
+	ETHERNET_CONFIG_TYPE_DUPLEX,
+	ETHERNET_CONFIG_TYPE_MAC_ADDRESS,
+};
+
+struct ethernet_config {
+/** @cond ignore */
+	union {
+		bool auto_negociation;
```

Comment:
Almost missed this one. s/negociation/negotiation/

Reply:
han, I was counting on checkpatch  (or some of the verification script) to check for my bad-english. Will fix that